### ISSUE TITLE
combineLatest without `project` returns mutating array.

### ISSUE THREAD

**Issue by Dorus** [ReactiveX/rxjs#1851]
We have been using `combineLatest` in combination with `distinctUntilChanged` and ran into an issue where the array returned from `combineLatst` was mutation and reused, resulting in `distinctUntilChanged` comparing the array against itself.

Unsafe:
```JS
console.clear();

let first$ =  Rx.Observable.interval(500).take(3).map(i => `A${i}`);
let second$ =  Rx.Observable.interval(500).take(3).map(i => `B${i}`);
let third$ =  Rx.Observable.interval(500).take(3).map(i => `C${i}`);

let combined$ = Rx.Observable
          .combineLatest(first$, second$, third$)
          .debounceTime(1)
          .flatMap(state => Rx.Observable.interval(10).take(2).map(i => state))
          .distinctUntilChanged(x=>x, (a, b) => {
            console.log("A:" + a);
            console.log("B:" + b);
            return a !== b;
          });

combined$.subscribe(console.log);
```

Safe:

```
console.clear();

let first$ =  Rx.Observable.interval(500).take(3).map(i => `A${i}`);
let second$ =  Rx.Observable.interval(500).take(3).map(i => `B${i}`);
let third$ =  Rx.Observable.interval(500).take(3).map(i => `C${i}`);

let combined$ = Rx.Observable
          .combineLatest(first$, second$, third$, (a, b, c) => [a, b, c]))
          .debounceTime(1)
          .flatMap(state => Rx.Observable.interval(10).take(2).map(i => state))
          .distinctUntilChanged(x=>x, (a, b) => {
            console.log("A:" + a);
            console.log("B:" + b);
            return a !== b;
          });

combined$.subscribe(console.log);
```

I would expect both snippets to give the same result, as the `project` function from `combineLatest` is optional.

Ps. @babeal brought up this snipped on gitter on [28 juli 2016 19:07](https://gitter.im/Reactive-Extensions/RxJS?at=579a3bbb1d6bf8244e29b44e).

**Comment by kwonoj** [comment]
I assume this is dupe of issue https://github.com/ReactiveX/rxjs/issues/1686 with PR https://github.com/ReactiveX/rxjs/pull/1841?

**Comment by Dorus** [comment]
Looks very similar.

**Comment by blesh** [comment]
Closed by #1841
